sbom/examples/container_image/from_catalog.py

```diff
@@ -0,0 +1,232 @@
+            "downloadLocation": "NOASSERTION",  # TODO: should this be set to the registry+repo?
+            "licenseDeclared": image_license,
+            "externalRefs": [],
+            "checksums": [
+                {
+                    "algorithm": "SHA256",
+                    "checksumValue": image_digest,
+                }
+            ],
+        }
+        for name, repo_url, tag in sorted(repos):
+            purl = (
+                f"pkg:oci/{name}@sha256%3A{image_index_digest}?"
+                f"arch={image['architecture']}&repository_url={repo_url}&tag={tag}"
+            )
+            ref = {
+                "referenceCategory": "PACKAGE-MANAGER",
+                "referenceType": "purl",
+                "referenceLocator": purl,
+            }
+            image_pkg["externalRefs"].append(ref)
+
+        packages.append(image_pkg)
+        relationships.append(
+            {
+                "spdxElementId": "SPDXRef-image-index",
+                "relationshipType": "CONTAINS",
+                "relatedSpdxElement": spdx_image_id,
+            }
+        )
+
+        for rpm in get_rpms(catalog_image_id):
+            rpm_purl = (
+                f"pkg:rpm/redhat/{rpm['name']}@{rpm['version']}-{rpm['release']}?"
+                # We don't have a way to find out which content set (RPM repo) an RPM came from,
+                # so we arbitrarily choose one here (assuming we have this mapping via RPM
+                # lockfiles or other means eventually).
+                f"arch={rpm['architecture']}&repository_id={content_sets[0]}"
+            )
+
+            srpm_nevr = rpm["srpm_nevra"].rstrip(".src")
+            n, ev, r = srpm_nevr.rsplit("-", maxsplit=2)
+            srpm_purl = (
+                f"pkg:rpm/redhat/{n}@{ev}-{r}?arch=src"
+                # Same note applies as above; also, we're assuming the SRPM for a given binary
+                # RPM is hosted in the same repo.
+                f"&repository_id={content_sets[0]}"
+            )
+
+            spdx_rpm_id = f"SPDXRef-{rpm['architecture']}-{rpm['name']}"
+            rpm_pkg = {
+                "SPDXID": spdx_rpm_id,
+                "name": rpm["name"],
+                "versionInfo": rpm["version"],
+                "supplier": "Organization: Red Hat",
+                "downloadLocation": "NOASSERTION",  # Unset on purpose; refer to RPM SBOM
+                "packageFileName": rpm["nvra"] + ".rpm",
+                "licenseDeclared": "NOASSERTION",  # Unset on purpose; refer to RPM SBOM
+                "externalRefs": [
+                    {
+                        "referenceCategory": "PACKAGE-MANAGER",
+                        "referenceType": "purl",
+                        "referenceLocator": rpm_purl,
+                    },
+                    {
+                        "referenceCategory": "PACKAGE-MANAGER",
+                        "referenceType": "purl",
+                        "referenceLocator": srpm_purl,
```

Comment:
I added a purl for the SRPM that this binary RPM belongs to, so that the link exists between both the binary component and the SRPM component, but perhaps it's not needed? Do we even have information about SRPMs during the build of a container image that contains just binary RPMs?

Reply:
I think we don't, and I think it's not needed.